### PR TITLE
Adds a conn.mark= operation for a connection

### PR DESCRIPTION
**Pull request by zwoop**

### CODE REVIEW

File: include/cripts/Connections.hpp [hunk 1]
```diff
@@ -170,6 +169,33 @@ class ConnBase
...
+    friend class ConnBase;
+
+    Mark()                       = default;
+    void operator=(const Mark &) = delete;
+
+    // Same here, no API in ATS to Get() the mark on a VC.
+    operator integer() const { return _val; }
+
+    void
+    operator=(int val)
```
Comment (cmcfarlen) [inline include/cripts/Connections.hpp:186]: Rule of three (or five)?  Need copy constructor and destructor?

**Comment by zwoop** [comment]
[approve ci fedora]

**Comment by zwoop** [comment]
[approve ci autest]

**Comment by bryancall** [comment]
[approve ci autest]

**Review by bneradt — APPROVED** [review]

**Comment by cmcfarlen** [comment]
Cherry-picked to v10.0.x

**Pull request merged**